python/lsst/ts/tunablelaser/csc.py

```diff
@@ -0,0 +1,185 @@
+                            self.model.TK6.display_temperature_register_2.register_value)
+                        self.temperature_topic.ldco48bp_temperature = float(
+                            self.model.LDCO48BP.display_temperature_register.register_value)
+                        self.temperature_topic.ldco48bp_temperature_2 = float(
+                            self.model.LDCO48BP.display_temperature_register_2.register_value)
+                        self.temperature_topic.ldco48bp_temperature_3 = float(
+                            self.model.LDCO48BP.display_temperature_register_3.register_value)
+                        self.temperature_topic.m_ldco48_temperature = float(
+                            self.model.M_LDCO48.display_temperature_register.register_value
+                        )
+                        self.temperature_topic.m_ldco48_temperature_2 = float(
+                            self.model.M_LDCO48.display_temperature_register_2.register_value
+                        )
+                        self.tel_wavelength.put(self.wavelength_topic)
+                        self.tel_temperature.put(self.temperature_topic)
+                await asyncio.sleep(self.frequency)
+        except Exception:
+            if self.summary_state is not salobj.State.FAULT:
+                self.fault(code=TunableLaser.LaserErrorCode.hw_cpu_error, report=f"Problem with hardware.")
+
+    def assert_substate(self, substates, action):
+        """Assert that the action is happening while in the PropagatingState.
+
+        Parameters
+        ----------
+        substates : `list`
+            A list of allowed states
+        action : `str`
+            The name of the command being sent.
+
+        Raises
+        ------
+        salobj.ExpectedError
+            Raised when an action is not allowed in a substate.
+
+        """
+        if self.detailed_state not in [TunableLaser.LaserDetailedState(substate) for substate in substates]:
+            raise salobj.ExpectedError(f"{action} not allowed in state {self.detailed_state!r}")
+
+    async def handle_summary_state(self):
+        if self.enabled_or_disabled:
+            if not self.model.connected:
+                self.model.connect()
+            if self.telemetry_task.done():
+                self.telemetry_task = asyncio.create_task(self.telemetry())
+        else:
+            if self.model.is_propgating:
+                self.model.stop_propagating()
+            self.model.disconnect()
+            self.telemetry_task.cancel()
+
+    async def do_changeWavelength(self, data):
+        """Change the wavelength of the laser.
+
+        Parameters
+        ----------
+        data
+        """
+        self.assert_enabled("changeWavelength")
+        try:
+            self.model.change_wavelength(data.wavelength)
+        except TimeoutError as te:
+            self.fault(code=TunableLaser.LaserErrorCode.timeout_error, msg=te.msg)
```

Comment:
`msg` is not an argument to the `fault` message -- you want `report` -- and `te.msg` will not be interesting. The report is intended to help others figure out what went wrong.

Please consider something along the lines of: `report="Timed out changing wavelength")